python/fastquant/backtest/backtest.py

```diff
@@ -246,29 +249,51 @@ class CustomData(bt.feeds.PandasData):
         print("Number of strat runs:", len(stratruns))
         print("Number of strats per run:", len(stratruns[0]))
         print("Strat names:", strat_names)
+
+    history_dfs = {}
     for stratrun in stratruns:
         strats_params = {}
 
         if verbose:
             print("**************************************************")
+
+        
         for i, strat in enumerate(stratrun):
+            strat_name = strat_names[i]
             p_raw = strat.p._getkwargs()
             p = {}
             for k, v in p_raw.items():
                 if k not in ["periodic_logging", "transaction_logging"]:
                     # Make sure the parameters are mapped to the corresponding strategy
                     if strategy == "multi":
                         key = (
-                            "{}.{}".format(strat_names[i], k)
+                            "{}.{}".format(strat_name, k)
                             if k not in GLOBAL_PARAMS
                             else k
                         )
+
+                        selected_p = {}
```

Comment:
Ah, `selected_p={}` should be outside the loop (in line 264). That's why only the last parameter from strats dict in backtest input is used as the final key. Anyway it is unique so it kinda works.

Reply:
@jpdeleon Right now, only one transaction table is returned even with the grid search functionality.

Can we make sure we have one transactions table for each strategy in the grid search? Since we're planning to merge the dataframes together, I think we can forego using a `dict` for `history_dfs` and can turn it into a `list` instead, which we will `pd.concat` later :smile:

```
from fastquant import *
df = get_stock_data("JFC", "2019-01-01", "2020-01-01")
res, trans = backtest("smac", df, slow_period=[40, 45], return_transactions=True)
print(trans)

{'smac':            buy_price sell_price      cost commision
 dt                                                 
 2019-04-03    317.00       None  98904.00    741.78
 2019-04-23      None     302.20  98904.00    707.15
 2019-07-17    284.60       None  93064.20    697.98
 2019-07-22      None     275.00  93064.20    674.44
 2019-10-17    233.00       None  88540.00    664.05
 2019-11-15      None     202.00  88540.00    575.70
 2019-12-20    216.00       None  75600.00    567.00}
```

Reply:
Ah I know why. I'll add another commit.